HDFEventOutputer.cc

```diff
@@ -0,0 +1,204 @@
+  ids.push_back(iEventID.event);
+  write_ds<unsigned long long>(gid, "Event_IDs", ids);
+  write_ds<char>(gid, "products", iBuffer);
+  std::vector<uint32_t> offset;
+  offset.resize(iOffsets.size()); 
+ // std::copy( iOffsets.begin(), iOffsets.end(), offset.begin() );
+  write_ds<uint32_t>(gid, "offsets", iOffsets); 
+}
+
+void 
+HDFEventOutputer::writeFileHeader(SerializeStrategy const& iSerializers) {
+  constexpr hsize_t ndims = 1;
+  constexpr hsize_t     dims[ndims] = {0};
+  constexpr hsize_t     chunk_dims[ndims] = {128};
+  constexpr hsize_t     max_dims[ndims] = {H5S_UNLIMITED};
+  auto space = hdf5::Dataspace::create_simple (ndims, dims, max_dims); 
+  auto prop   = hdf5::Property::create();
+  prop.set_chunk(ndims, chunk_dims);
+  hdf5::Group g = hdf5::Group::create(file_, "Lumi");
+  hdf5::Dataset::create<int>(g, "Event_IDs", space, prop);
+  hdf5::Dataset::create<char>(g, "products", space, prop);
+  hdf5::Dataset::create<int>(g, "offsets", space, prop);
+
+  const auto scalar_space  = hdf5::Dataspace::create_scalar();
+  hdf5::Attribute::create<int>(g, "run", scalar_space);
+  hdf5::Attribute::create<int>(g, "lumisec", scalar_space);
+  
+  for(auto const& s: iSerializers) {
+    std::string const type(s.className());
+    std::string const name(s.name());
+    constexpr hsize_t     str_dims[ndims] = {10};
+    auto const attr_type = H5Tcopy (H5T_C_S1);
+    H5Tset_size(attr_type, H5T_VARIABLE);
+    auto const attr_space  = H5Screate(H5S_SCALAR);
+    hdf5::Attribute prod_name = hdf5::Attribute::create<std::string>(g, name.c_str(), attr_space); 
+    prod_name.write<std::string>(type);
+  }
+}
+
+std::pair<std::vector<uint32_t>, std::vector<char>> HDFEventOutputer::writeDataProductsToOutputBuffer(SerializeStrategy const& iSerializers) const{
+  //Calculate buffer size needed
+  uint32_t bufferSize = 0;
+  std::vector<uint32_t> offsets;
+  offsets.reserve(iSerializers.size()+1);
+  for(auto const& s: iSerializers) {
+    auto const blobSize = s.blob().size();
+    offsets.push_back(bufferSize);
+    bufferSize += blobSize;
+  }
+  offsets.push_back(bufferSize);
+
+  //initialize with 0
+  std::vector<char> buffer(bufferSize, 0);
+  
+  {
+    uint32_t index = 0;
+    for(auto const& s: iSerializers) {
+      auto offset = offsets[index++];
+      std::copy(s.blob().begin(), s.blob().end(), buffer.begin()+offset );
+    }
+    assert(buffer.size() == offset[index]);
+  }
+
+  return {offsets, buffer};
+}
+namespace {
+  class HDFEventMaker : public OutputerMakerBase {
+  public:
+    HDFEventMaker(): OutputerMakerBase("HDFEventOutputer") {}
+    
+    std::unique_ptr<OutputerBase> create(unsigned int iNLanes, ConfigurationParameters const& params) const final {
+      auto fileName = params.get<std::string>("fileName");
+      if(not fileName) {
+        std::cout<<" no file name given for HDFEventOutputer\n";
+        return {};
+      }
+
+      auto batchSize = params.get<int>("batchSize", 1);
```

Comment:
Since this implementation does not support batch size, why offer the parameter?

Reply:
yes, replacing it with chunk size.